spring-kafka/src/test/java/org/springframework/kafka/core/reactive/ReactiveKafkaProducerTemplateTransactionIntegrationTests.java

```diff
@@ -0,0 +1,288 @@
+	private static final Duration DEFAULT_VERIFY_TIMEOUT = Duration.ofSeconds(10);
+
+	@ClassRule
+	public static KafkaEmbedded embeddedKafka = new KafkaEmbedded(3, true, DEFAULT_PARTITIONS_COUNT, REACTIVE_INT_KEY_TOPIC);
+
+	private static ReactiveKafkaConsumerTemplate<Integer, String> reactiveKafkaConsumerTemplate;
+	private ReactiveKafkaProducerTemplate<Integer, String> reactiveKafkaProducerTemplate;
+
+	@BeforeClass
+	public static void setUpBeforeClass() {
+		Map<String, Object> consumerProps = KafkaTestUtils.consumerProps("reactive_transaction_consumer_group", "false", embeddedKafka);
+		reactiveKafkaConsumerTemplate = new ReactiveKafkaConsumerTemplate<>(setupReceiverOptionsWithDefaultTopic(consumerProps));
+	}
+
+	@Before
+	public void setUp() {
+		reactiveKafkaProducerTemplate = new ReactiveKafkaProducerTemplate<>(setupSenderOptionsWithDefaultTopic(), new MessagingMessageConverter());
+	}
+
+	private SenderOptions<Integer, String> setupSenderOptionsWithDefaultTopic() {
+		Map<String, Object> senderProps = KafkaTestUtils.senderProps(embeddedKafka.getBrokersAsString());
+		SenderOptions<Integer, String> senderOptions = SenderOptions.create(senderProps);
+		senderOptions = senderOptions
+				.producerProperty(ProducerConfig.TRANSACTIONAL_ID_CONFIG, "reactive.transaction")
+				.producerProperty(ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG, true)
+				.producerProperty(ProducerConfig.RETRIES_CONFIG, 1);
+		return senderOptions;
+	}
+
+	private static ReceiverOptions<Integer, String> setupReceiverOptionsWithDefaultTopic(Map<String, Object> consumerProps) {
+		ReceiverOptions<Integer, String> basicReceiverOptions = ReceiverOptions.create(consumerProps);
+		return basicReceiverOptions
+				.consumerProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+				.consumerProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false")
+				.consumerProperty(ConsumerConfig.ISOLATION_LEVEL_CONFIG, "read_committed")
+				.addAssignListener(p -> Assertions.assertThat(p.iterator().next().topicPartition().topic()).isEqualTo(REACTIVE_INT_KEY_TOPIC))
+				.subscription(Collections.singletonList(REACTIVE_INT_KEY_TOPIC));
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		reactiveKafkaProducerTemplate.close();
+	}
+
+	@Test
+	public void shouldSendOneRecordTransactionallyViaTemplateAsSenderRecordAndReceiveIt() {
+		ProducerRecord<Integer, String> producerRecord =
+				new ProducerRecord<>(REACTIVE_INT_KEY_TOPIC, DEFAULT_PARTITION, DEFAULT_TIMESTAMP, DEFAULT_KEY, DEFAULT_VALUE);
+		long correlationMetadata = 1L;
+		SenderRecord<Integer, String, Long> senderRecord = SenderRecord.create(producerRecord, correlationMetadata);
+
+		Mono<SenderResult<Long>> publisher = reactiveKafkaProducerTemplate.sendTransactionally(senderRecord);
+		StepVerifier.create(publisher)
+				.assertNext(senderResult ->
+						Assertions.assertThat(senderResult.recordMetadata())
+								.has(match(recordMetadata -> REACTIVE_INT_KEY_TOPIC.equals(recordMetadata.topic())))
```

Comment:
I somehow believe that we can live without extra `match()` utility.
There is a `hasFieldOrPropertyWithValue()` and its siblings.

Reply:
Done. Replaced by `extracting+containsExactly` and `allSatisfy`.